### PR TITLE
Task01

### PR DESCRIPTION
**Pull request by ando-andy**
## Related Tickets

- [#Ticket ID](https://dev.framgia.com/redmine/issues/???)

## What's this PR do ?
create TabBar

## Library
*(List gem, library third party add new)*

CleanArchitecture
'MGArchitecture', '0.4.0'
'MGAPIService', '0.5.0'
'MGLoadMore', '0.3.0'

Core
'ObjectMapper', '3.3'
'Reusable', '4.0.4'
'Then', '2.4'
'MJRefresh', '3.1'
'Validator', '3.0.2'

Rx
'NSObject+Rx', '4.4'
'RxDataSources', '3.1'

Core Data
'MagicalRecord', '2.3'

'OrderedSet', '3.0'
'MBProgressHUD', '1.1'
'SDWebImage', '5.0'
'ActionSheetPicker-3.0', '2.3'

test
'RxBlocking', '4.5'

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*
TabBar created

## Checklist

- [x] It was tested in local success ?
- [x] Updated rake task, environment variable at [here](https://???)
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [ ] Validate UI/Model/API

## Notes
*(Other notes)*
https://docs.google.com/spreadsheets/d/1F5V2sIUMGej3Wrp8qYRscf0KaVSlccBmqkRx49_hUVo/edit#gid=935273790

### CODE REVIEW

File: MovieDB/Scenes/Main/MainNavigator.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+protocol MainNavigatorType {
+    func toMain()
+
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Main/MainNavigator.swift]: remove line

File: MovieDB/Assembler/Assembler.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  Created by kazutaka.ando on 2019/06/10.
+//  Copyright © 2019 Kazando. All rights reserved.
+//
+
+protocol Assembler: class,
+    MovieAssembler,
+    MainAssembler,
+    CategoriesAssembler,
+    FavoriteAssembler,
+AppAssembler {
```
Comment (tgdong2296) [inline MovieDB/Assembler/Assembler.swift]:
```suggestion
     AppAssembler {
```

File: MovieDBTests/Scenes/Categories/CategoriesViewControllerTests.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    override func setUp() {
+        super.setUp()
+        viewController = CategoriesViewController.instantiate()
+    }
+
+    func test_ibOutlets() {
+        _ = viewController.view
+        XCTAssert(true)
+//        XCTAssertNotNil(viewController.tableView)
```
Comment (thanhlong12312) [inline MovieDBTests/Scenes/Categories/CategoriesViewControllerTests.swift]: remove comment line

File: MovieDBTests/Scenes/Favorite/FavoriteViewControllerTests.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    override func setUp() {
+        super.setUp()
+        viewController = FavoriteViewController.instantiate()
+    }
+
+    func test_ibOutlets() {
+        _ = viewController.view
+        XCTAssert(true)
+//        XCTAssertNotNil(viewController.tableView)
```
Comment (thanhlong12312) [inline MovieDBTests/Scenes/Favorite/FavoriteViewControllerTests.swift]: remove comment line

File: MovieDBTests/Scenes/Movie/MovieViewControllerTests.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    override func setUp() {
+        super.setUp()
+        viewController = MovieViewController.instantiate()
+    }
+
+    func test_ibOutlets() {
+        _ = viewController.view
+        XCTAssert(true)
+//        XCTAssertNotNil(viewController.tableView)
```
Comment (thanhlong12312) [inline MovieDBTests/Scenes/Movie/MovieViewControllerTests.swift]: remove

File: MovieDBTests/Scenes/Main/MainViewControllerTests.swift [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    var viewController: MainViewController!
+
+    override func setUp() {
+        super.setUp()
+        viewController = MainViewController()
+    }
+
+    func test_ibOutlets() {
+        XCTAssertNotNil(viewController)
+
```
Comment (tgdong2296) [inline MovieDBTests/Scenes/Main/MainViewControllerTests.swift]: delete this line

File: MovieDBUITests/MovieDBUITests.swift [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+//  MovieDBUITests
+//
+//  Created by kazutaka.ando on 2019/06/10.
+//  Copyright © 2019 Kazando. All rights reserved.
+//
+
+import XCTest
+@testable import MovieDB
+
+class MovieDBUITests: XCTestCase {
```
Comment (tgdong2296) [inline MovieDBUITests/MovieDBUITests.swift]: delete this file because you don't need to use

File: MovieDB/Scenes/Main/MainNavigator.swift (hunk 1 above)
Comment (ando-andy) [reply MovieDB/Scenes/Main/MainNavigator.swift]: Removed

File: MovieDBTests/Scenes/Categories/CategoriesViewControllerTests.swift (hunk 3 above)
Comment (ando-andy) [reply MovieDBTests/Scenes/Categories/CategoriesViewControllerTests.swift]: Removed

File: MovieDBTests/Scenes/Favorite/FavoriteViewControllerTests.swift (hunk 4 above)
Comment (ando-andy) [reply MovieDBTests/Scenes/Favorite/FavoriteViewControllerTests.swift]: Removed

File: MovieDBTests/Scenes/Movie/MovieViewControllerTests.swift (hunk 5 above)
Comment (ando-andy) [reply MovieDBTests/Scenes/Movie/MovieViewControllerTests.swift]: Removed

File: MovieDB/Assembler/Assembler.swift (hunk 2 above)
Comment (ando-andy) [reply MovieDB/Assembler/Assembler.swift]: Modified

File: MovieDBTests/Scenes/Main/MainViewControllerTests.swift (hunk 6 above)
Comment (ando-andy) [reply MovieDBTests/Scenes/Main/MainViewControllerTests.swift]: Deleted

File: MovieDBUITests/MovieDBUITests.swift (hunk 7 above)
Comment (ando-andy) [reply MovieDBUITests/MovieDBUITests.swift]: Deleted UITest

**Pull request merged**